helpers/search_helper.py

```diff
@@ -55,6 +56,31 @@ def update_team_location_index(cls, team):
     def remove_team_location_index(cls, team):
         search.Index(name=cls.TEAM_LOCATION_INDEX).delete(team.key.id())
 
+    @classmethod
+    def update_match_number_index(cls, match):
+        if match.year:
+            fields = [
+                search.NumberField(name='year', value=match.year),
+                search.NumberField(name='match_number', value=match.match_number),
+                search.NumberField(name='set_number', value=match.set_number),
+                search.NumberField(name='comp_level_sort', value=match.comp_level_sort),
+                search.NumberField(name='num_videos', value=match.num_videos),
+                search.TextField(name='comp_level', value=match.comp_level),
+                search.TextField(name='event_key', value=match.event_key),
+                search.TextField(name='team1', value=match.team1),
+                search.TextField(name='team2', value=match.team2),
+                search.TextField(name='team3', value=match.team3),
+                search.TextField(name='team4', value=match.team4),
+                search.TextField(name='team5', value=match.team5),
+                search.TextField(name='team6', value=match.team6),
+            ]
+            search.Index(name="matchNumber").put(
+                search.Document(doc_id=match.key.id(), fields=fields))
+
+    @classmethod
+    def remove_match_number_index(cls, match):
```

Comment:
Similar to above, a task that enqueues a call to this should be added to MatchManipulator.postDeleteHook() to keep the search index in sync with this. Add to the queue search-index-update.